app/models/preorder_information.rb

```diff
@@ -42,6 +42,10 @@ def infer_status
     end
   end
 
+  def refresh_status
```

Comment:
minor niggle - can this method be `refresh_status!` as it both saves state, and can throw an exception?

Reply:
fair - done!